CONTRIBUTING.md

```diff
@@ -36,6 +36,25 @@ Coding Standards]. Compliance of your code is checked automatically using
 Travis CI. See [clang-format] and [clang-tidy] configs for more details about
 coding standards.
 
+## How to add an extension
+
+First of all please make sure you have added a link to the specification for the
+extension in your PR. Then to add definitions of new Op Codes you shall modify
+[spirv.hpp] which is an external dependency for this project. To do so you
+should add new definitions to [json grammar file], rebuild the header following
+the [instructions] in [SPIR-V Headers repository] and push your changes for
+review, i.e. make a PR. Once the PR is merged a brand-new spirv.hpp will be
+downloaded during build of the translator.
+
+It's highly recommended to add the definitions to [SPIR-V Headers repository]
+first, but if you don't want to bring it there yet, you can define new Op Codes
+in the [internal SPIR-V header file].
+
+For local testing you can copy spirv.hpp to {PATH}/include/spirv/unified1 and
+modify it. To use it for your local build please specify
+EXTERNAL_SPIRV_HEADERS_DIR cmake variable with your {PATH} to build the
```

Comment:
`EXTERNAL_SPIRV_HEADERS_DIR` is not referenced in `CMakeLists.txt` anymore, the doc should be updated

Reply:
Done.